test/test_command.py

```diff
@@ -1,65 +1,46 @@
     p = parser_create()
+    with pytest.raises(SystemExit) as sys_exit:
+        p.parse_args([])
+    assert sys_exit.type == SystemExit
+    assert sys_exit.value.code == 2
+
+    captured_stderr = capsys.readouterr().err
+    assert "the following arguments are required: command" in captured_stderr
+
+
+@pytest.mark.parametrize("command,results,sys_exit_code,sys_exit_message", [
+    (["compile"], {"command": "compile", "output": None, "input": None}, None, None),
+    (["compile", "foo"], {"command": "compile", "output": None, "input": "foo"}, None, None),
+    (["compile", "-o", "output_manifest.yaml"], {"command": "compile", "output": "output_manifest.yaml", "input": None}, None, None),
+    (["compile", "-o", "output_manifest.yaml", "input_omifest.yaml"]
+     , {"command": "compile", "output": "output_manifest.yaml", "input": "input_omifest.yaml"}, None, None),
+    (["compile", "--no_such_option"], {}, 2, "unrecognized arguments: --no_such_option"),
+
+    (["validate", "-o", "output_manifest.yaml", "input_omifest.yaml"]
+     , {}, 2
+     # deliberately we expect '-o input_omifest.yaml' as argparse acts like this if -o is not defined
+     , "error: unrecognized arguments: -o input_omifest.yaml"),
+    (["validate", "foo.yaml"], {"command": "validate", "input": "foo.yaml"}, None, None),
+    ]
+)
+def test_parse_commands(capsys, command, results, sys_exit_code, sys_exit_message):
 
-    r = p.parse_args([])
-
-    assert r.command is None
-    assert r.verbose == 0
-    assert not r.json
-    assert r.identifier is None
-
-    r = p.parse_args(["-j"])
-    assert r.command is None
-    assert r.json
-    assert r.verbose == 0
-    assert r.identifier is None
-
-    r = p.parse_args(["-j", "-v"])
-    assert r.command is None
-    assert r.json
-    assert r.verbose == 1
-    assert r.identifier is None
-
-    r = p.parse_args(["-j", "-vvvv"])
-    assert r.command is None
-    assert r.json
-    assert r.verbose == 4
-    assert r.identifier is None
-
-    r = p.parse_args(["--json", "--verbose", "--verbose"])
-    assert r.command is None
-    assert r.json
-    assert r.verbose == 2
-    assert r.identifier is None
-
-    r = p.parse_args(["-i", "foo"])
-    assert r.command is None
-    assert not r.json
-    assert r.verbose == 0
-    assert r.identifier == "foo"
-
-
-def test_parse_compile():
     p = parser_create()
-
-    r = p.parse_args(["compile"])
-    assert r.command == "compile"
-    assert r.output is None
-
-    r = p.parse_args(["compile", "foo"])
-    assert r.command == "compile"
-    assert r.input == "foo"
-    assert r.output is None
-
-    r = p.parse_args(["compile", "-o", "file.yaml"])
-    assert r.command == "compile"
-    assert r.input is None
-    assert r.output == "file.yaml"
-
-    r = p.parse_args(["compile", "-o", "file.yaml", "foo.yaml"])
-    assert r.command == "compile"
-    assert r.input == "foo.yaml"
-    assert r.output == "file.yaml"
+    if sys_exit_code:
```

Comment:
(super nitpick) Given how small the body is, I wonder if it's worthwhile to split the test into two, one "happy" and one "sad" case. This way the inputs are slightly smaller and the pytest.mark.parametriz might be slightly easier to read (or maybe not) - in any case, more of a thought then aynthing else, fine as is.